oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -332,13 +329,21 @@ private void scan() {
                 LOG.trace("Kicking new search after query {}", searchRequest.source());
 
                 searchStartTime = System.currentTimeMillis();
+                requests++;
                 indexNode.getConnection().getClient().searchAsync(searchRequest, RequestOptions.DEFAULT, this);
                 elasticMetricHandler.markQuery(false);
             } else {
                 LOG.trace("Scanner is closing or still processing data from the previous scan");
             }
         }
 
+        /* picks the size in the fetch array at index=requests or the last if out of bound */
+        private int getFetchSize() {
```

Comment:
Maybe rename to "getCurrentFetchSize()"? To make it clear that the fetch size is variable... Or (possibly better) change signature to "getFetchSize(int requestId)", that would also make it very clear.

Reply:
Makes sense. Done.